components/core/tests/test-encoding_methods.cpp

```diff
@@ -226,61 +226,43 @@ TEMPLATE_TEST_CASE("Encoding floats", "[ffi][encode-float]", eight_byte_encoded_
     decoded_value = decode_float_var(encoded_var);
     REQUIRE(decoded_value == value);
 
-    // Test non-floats
-    value = "";
-    REQUIRE(!encode_float_string(value, encoded_var));
-
-    value = "a";
-    REQUIRE(!encode_float_string(value, encoded_var));
-
-    value = "-";
-    REQUIRE(!encode_float_string(value, encoded_var));
-
-    value = "+";
-    REQUIRE(!encode_float_string(value, encoded_var));
-
-    value = "-a";
-    REQUIRE(!encode_float_string(value, encoded_var));
-
-    value = "+a";
-    REQUIRE(!encode_float_string(value, encoded_var));
-
-    value = "--";
-    REQUIRE(!encode_float_string(value, encoded_var));
-
-    value = "++";
-    REQUIRE(!encode_float_string(value, encoded_var));
-
-    // Test unrepresentable values
-    value = ".";
-    REQUIRE(!encode_float_string(value, encoded_var));
-
-    value = "1.";
-    REQUIRE(!encode_float_string(value, encoded_var));
-
-    value = " 1.0";
-    REQUIRE(!encode_float_string(value, encoded_var));
-
-    value = "- 1.0";
-    REQUIRE(!encode_float_string(value, encoded_var));
-
-    value = "1.0 ";
-    REQUIRE(!encode_float_string(value, encoded_var));
-
-    value = "+1.0";
-    REQUIRE(!encode_float_string(value, encoded_var));
-
-    value = "1.0f";
-    REQUIRE(!encode_float_string(value, encoded_var));
-
-    value = "1.0F";
-    REQUIRE(!encode_float_string(value, encoded_var));
-
-    value = "1.0l";
-    REQUIRE(!encode_float_string(value, encoded_var));
+    // Test unrepresentable floating values
```

Comment:
Because `GENERATE` is multiplicative in the same control flow, we should wrap it in a `SECTION` to limit its scope:

```suggestion
    SECTION("Test unrepresentable floats") {
```